nessus-agent/elx/init.sls

```diff
@@ -30,6 +30,75 @@ Print nessus-agent help:
         without linking the agent to the server.
 {%- endif %}
 
+{%- if (nessus.log_rotation_strategy) %}
+Add Log Rotation Strategy:
+  file.replace:
+    - name: '/opt/nessus_agent/var/nessus/log.json'
+    - pattern: '"file": "/opt/nessus_agent/var/nessus/logs/www_server.log"'
+    - repl: '"rotation_strategy": {{nessus.log_rotation_strategy}},\n\t\t\g<0>'
+    - watch:
+      - cmd: Pause For Log File
+{%- endif %}
+
+{%- if (nessus.log_rotation_time) %}
+Add Log Rotation Time:
+  file.replace:
+    - name: '/opt/nessus_agent/var/nessus/log.json'
+    - pattern: '"file": "/opt/nessus_agent/var/nessus/logs/www_server.log"'
+    - repl: '"rotation_time": {{nessus.log_rotation_time}},\n\t\t\g<0>'
+    - watch:
+      - cmd: Pause For Log File
+{%- endif %}
+
+{%- if (nessus.log_max_size) %}
+Add Log Max Size:
+  file.replace:
+    - name: '/opt/nessus_agent/var/nessus/log.json'
+    - pattern: '"file": "/opt/nessus_agent/var/nessus/logs/www_server.log"'
+    - repl: '"max_size": {{nessus.log_max_size}},\n\t\t\g<0>'
+    - watch:
+      - cmd: Pause For Log File
+{%- endif %}
+
+{%- if (nessus.log_max_files) %}
+Add Log Max Files:
+  file.replace:
+    - name: '/opt/nessus_agent/var/nessus/log.json'
+    - pattern: '"file": "/opt/nessus_agent/var/nessus/logs/www_server.log"'
+    - repl: '"max_files": {{nessus.log_max_files}},\n\t\t\g<0>'
+    - watch:
+      - cmd: Pause For Log File
+{%- endif %}
+
+Pause For Log File:
+  cmd.run:
+    - name: sleep 5
+    - watch:
+      - cmd: Start Nessus Agent
+
+Pre-Create Nessus Log Directory:
+  file.directory:
+    - name: /var/log/nessus/logs
+    - user: root
+    - group: root
+    - dir_mode: 0755
+    - recurse:
+      - user
+      - group
+      - mode
+    - makedirs: True
+
+Create Sym-link To Log Dir:
+  file.symlink:
+    - name: /opt/nessus_agent/var/nessus/logs
```

Comment:
Does this create `/var/log/nessus/logs` as a symlink, or `/opt/nessus_agent/var/nessus/logs`? I think the latter is the intention of the open issue... The goal is for the `/var/log` partition to contain the actual log files...

Reply:
I believe this does the latter.

The `logs` symlink is created in `/opt/nessus_agent/var/nessus/` and points to `/var/log/nessus/logs`.  The files created when the agent is installed are stored in `/var/log/nessus/logs`.

This is what is shown in `/opt/nessus_agent/var/nessus/`:

![nessus agent log folder](https://user-images.githubusercontent.com/31355535/41621656-4d3a4cfc-73db-11e8-9dde-23335aa91f3d.png)

Reply:
Alright, good deal, that's what I wanted to verify. :D